src/test/java/seedu/address/logic/parser/AddLinkCommandParserTest.java

```diff
@@ -31,15 +36,38 @@ public void parse_invalidArgs_throwsParseException() {
                 AddLinkCommand.MESSAGE_USAGE));
     }
 
-    //Specific testing of link compatability performed at ParserUtilTest and LinkTest
+    @Test
+    public void parse_invalidArgsEmptyModuleCode_throwsParseException() {
+        assertParseFailure(parser, " " + PREFIX_MODULE_CODE, ModuleCode.MESSAGE_CONSTRAINTS);
+    }
+
+    @Test
+    public void parse_invalidArgsEmptyModuleUrlAndAliasWithFlags_throwsParseException() {
+        assertParseFailure(parser, " " + PREFIX_MODULE_CODE + VALID_MODULE_CODE + " "
+                + PREFIX_MODULE_LINK_URL + " " + PREFIX_MODULE_LINK_ALIAS + " ",
+                Link.MESSAGE_CONSTRAINTS_ALIAS);
+    }
+
+    //Specific testing of link alias/url validity performed at ParserUtilTest and LinkTest ==========
+    //Integration testing of AddLinkCommandParser with ParserUtil
+    @Test
+    public void parse_invalidArgsOnlyWhiteSpace_throwsParseException() {
+        assertParseFailure(parser, " " + PREFIX_MODULE_CODE + VALID_MODULE_CODE + " "
+                + PREFIX_MODULE_LINK_URL + "              " + PREFIX_MODULE_LINK_ALIAS + "            ",
+                Link.MESSAGE_CONSTRAINTS_ALIAS);
+    }
+
     //Integration testing of AddLinkCommandParser with ParserUtil
     @Test
     public void parse_links_success() {
```

Comment:
```suggestion
    public void parse_validLinks_success() {
```
Would this naming be clearer?